media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -587,12 +587,26 @@ bool MediaPipelineServerInternal::haveDataInternal(MediaSourceStatus status, uin
         return true;
     }
     m_activeRequests->erase(needDataRequestId);
+
+    int &counter = m_attachedSourcesForCounter[mediaSourceType];
+
     if (status != MediaSourceStatus::OK && status != MediaSourceStatus::EOS)
     {
-        RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status", needDataRequestId);
+        counter++;
+        if (counter == 10)
```

Comment:
magic number, define constexpr int kMaxSkippedNoAvailableSamples. Furthermore, we should skip only NO_AVAILABLE_SAMPLES. Other errors, like ERROR or CODEC_CHANGED have to be printed every time